### PR TITLE
Clipboard copy

### PR DESCRIPTION
**Pull request by hosseintoussi**
adds the ability of copying a code-block into clipboard

### CODE REVIEW

File: app/assets/javascripts/clipboard.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+$(document).on("turbolinks:load", function () {
+  addCopyButton();
+});
+
+function addCopyButton() {
+  new Clipboard('.clipboard-copy');
+
+  $("pre").each(function(index) {
+    var id = "id_" + index;
+    $(this).children("code").attr("id", id);
```
Comment (nomorejay) [inline app/assets/javascripts/clipboard.js]: you may want to assign $(this) to a variable:
`$this = $(this)`
that way it caches the fetch from the DOM for all three calls.

File: app/assets/javascripts/clipboard.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    prependButtonAndHide($(this), id);
+    toggleCopyButton($(this));
+  });
+}
+
+function prependButtonAndHide(pre, id) {
+  if (pre.find(".clipboard-copy").length == 0) {
```
Comment (nomorejay) [inline app/assets/javascripts/clipboard.js]: should use triple equals when possible:
`if (pre.find(".clipboard-copy").length === 0)`

File: app/assets/javascripts/clipboard.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    var button = $("<div class='ui small basic icon buttons right floated'>" +
```
Comment (nomorejay) [inline app/assets/javascripts/clipboard.js]: it's been a while but i forget if you actually need to wrap this in jQuery. you might be able to just prepend `button` as a string but not sure.

File: app/assets/javascripts/clipboard.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  $("pre").each(function(index) {
+    var id = "id_" + index;
+    $(this).children("code").attr("id", id);
+    prependButtonAndHide($(this), id);
+    toggleCopyButton($(this));
+  });
+}
+
+function prependButtonAndHide(pre, id) {
```
Comment (nomorejay) [inline app/assets/javascripts/clipboard.js]: it's convention to name variables that are jquery objects with a dollar sign in front of them `$pre` but not necessary.

**Comment by ricardovaleriano** [comment]
Nice work and a great addition, no doubt! =)

Since @nomorejay is killing on the PR, I'll shut up. But first I will scream: :shipit: ❗️

**Comment by hosseintoussi** [comment]
Thanks guys. Much appreciated.

**Pull request merged**